modules/videoio/src/cap_avfoundation.mm

```diff
@@ -1255,13 +1255,15 @@ -(int) updateImage {
         //exception;
     }
 
-    // Two codec supported AVVideoCodecH264 AVVideoCodecJPEG
+    // Three codec supported AVVideoCodecH264 AVVideoCodecJPEG AVVideoCodecTypeHEVC
     // On iPhone 3G H264 is not supported.
     if (fourcc == CV_FOURCC('J','P','E','G') || fourcc == CV_FOURCC('j','p','e','g') ||
             fourcc == CV_FOURCC('M','J','P','G') || fourcc == CV_FOURCC('m','j','p','g') ){
         codec = [AVVideoCodecJPEG copy]; // Use JPEG codec if specified, otherwise H264
     }else if(fourcc == CV_FOURCC('H','2','6','4') || fourcc == CV_FOURCC('a','v','c','1')){
             codec = [AVVideoCodecH264 copy];
+    }else if(fourcc == CV_FOURCC('H','E','V','C') || fourcc == CV_FOURCC('h','e','v','c') || fourcc == CV_FOURCC('H','2','6','5')){
+            codec = [AVVideoCodecTypeHEVC copy];
```

Comment:
Please add proper version check for that.

Reply:
Sorry, version of what should I check? iOS?

Reply:
Thanks to @VadimLevin